### ISSUE TITLE
Type-slice shortcut for choices in R4 does not yet work

### ISSUE THREAD

**Issue by ewoutkramer** [ewoutkramer/fhir-net-api#824]
In R4, you can introduce constraints for a specific type in a choice type, without using a type slice like so:

```
path Observation.value[x]
   ... constraints on all types, if any, e.g. cardinality

path Observation.valueQuantity
   ... constraints on values of type Quantity
```

These two element paths may now both exist in a set of ElementDefinitions. Before R4, you could have only one of those, and it would mean that value[x] was constrained to *only* allow Quantities.  We have not yet implemented having this combination. Not in the snapshot generator, nor in the validator.

**Comment by wmrutten** [comment]
I'm now working on updating the Snapshot Generator to support the new R4 shortcut syntax.

**Comment by ewoutkramer** [comment]
Fixed by #862
